src/seedu/addressbook/data/person/PrivatableContactDetail.java

```diff
@@ -0,0 +1,21 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * 
+ * Represents a contact detail that can be private
+ *
+ */
+public abstract class PrivatableContactDetail extends ContactDetail {
```

Comment:
I implemented it this way so that the code in getAsTextHiddenPrivate() is more general and easier to read
```
/**
* Formats a person as text, showing only non-private contact details.
*/
    default String getAsTextHidePrivate() {
        ArrayList<Printable> notPrivatePrintables = new ArrayList<Printable>();
        for (Printable contactDetail : getReadableContactDetails()) {
            if (!(contactDetail instanceof PrivatableContactDetail
                    && ((PrivatableContactDetail) contactDetail).isPrivate())) {
                notPrivatePrintables.add(contactDetail);
            }
        }

        Printable[] toPrint = new Printable[notPrivatePrintables.size()];
        toPrint = notPrivatePrintables.toArray(toPrint);
        return getPrintableString(toPrint);
    }
```